lib/cerner/oauth1a/access_token.rb

```diff
@@ -1,70 +1,138 @@
-      #                                responding to to_i that represents the expiration moment
-      #                                as the number of seconds since the epoch.
-      #             :nonce           - The String representing the nonce.
-      #             :expires_at      - A Time representing the creation moment or any object
-      #                                responding to to_i that represents the creation moment
-      #                                as the number of seconds since the epoch.
-      #             :token           - The String representing the token.
-      #             :token_secret    - The String representing the token secret.
-      #
-      # Raises ArgumentError if any of the arguments is nil
-      def initialize(accessor_secret:, consumer_key:, expires_at:, nonce:, timestamp:, token:, token_secret:)
-        raise ArgumentError, 'accessor_secret is nil' unless accessor_secret
+      #             :accessor_secret  - The optional String representing the accessor secret.
+      #             :consumer_key     - The required String representing the consumer key.
+      #             :expires_at       - An optional Time representing the expiration moment or any
+      #                                 object responding to to_i that represents the expiration
+      #                                 moment as the number of seconds since the epoch.
+      #             :nonce            - The required String representing the nonce.
+      #             :timestamp        - A required Time representing the creation moment or any
+      #                                 object responding to to_i that represents the creation
+      #                                 moment as the number of seconds since the epoch.
+      #             :token            - The required String representing the token.
+      #             :token_secret     - The required String representing the token secret.
+      #             :signature_method - The optional String representing the signature method.
+      #                                 Defaults to PLAINTEXT.
+      #             :signature        - The optional String representing the signature.
+      #                                 Defaults to nil.
+      #
+      # Raises ArgumentError if consumer_key, nonce, timestamp, token or signature_method is nil.
+      def initialize(
+        accessor_secret: nil,
+        consumer_key:,
+        expires_at: nil,
+        nonce:,
+        timestamp:,
+        token:,
+        token_secret: nil,
+        signature_method: 'PLAINTEXT',
+        signature: nil
+      )
         raise ArgumentError, 'consumer_key is nil' unless consumer_key
-        raise ArgumentError, 'expires_at is nil' unless expires_at
         raise ArgumentError, 'nonce is nil' unless nonce
+        raise ArgumentError, 'signature_method is nil' unless signature_method
```

Comment:
Since this just raises an error if someone passes in `nil`, thus overriding the default of `PLAINTEXT`, what if we just default it to `nil` in the parameter definition above and when setting it in the function do `@signature_method = signature_method || 'PLAINTEXT'`. Doing it this way eliminates one way to get an `ArgumentError`.

Reply:
I can do that, but I'm going to leave the 'PLAINTEXT' in the keyword argument, as I like how it documents the default behavior.

Reply:
Fixed at 26db3fb